File: union_types.cpp
#include <iostream>
using namespace std;
union Number { int i; float f; };
int main() {
    Number n;
    n.i = 5;
    cout << n.i << endl;
    n.f = 3.14;
    cout << n.f << endl;
    return 0;
}

File: union_types.rs
fn main() {
    enum Number { I(i32), F(f32) }
    let n1 = Number::I(5);
    let n2 = Number::F(3.14);
    match n1 { Number::I(i)=>println!("{}",i), Number::F(f)=>println!("{}",f) }
    match n2 { Number::I(i)=>println!("{}",i), Number::F(f)=>println!("{}",f) }
}
